src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -552,9 +576,119 @@ private static ITaskSeriesTimer StartParameterLogTimer(IRecurrentCommand updateC
+                {
+                    throw invokeException;
+                }
+            }
+
+            return invokeTask;
+        }
+
+
+        private static List<IFunctionInvocationFilter> GetFilters(FunctionDescriptor functionDescriptor, object instance)
+        {
+            List<IFunctionInvocationFilter> filters = new List<IFunctionInvocationFilter>();
+
+            if (instance != null)
+            {
+                IFunctionInvocationFilter instanceFilter = instance as IFunctionInvocationFilter;
+
+                if (instanceFilter != null)
+                {
+                    filters.Add(instanceFilter);
+                }
+            }
+            
+            // Check for method filters
+            if (functionDescriptor != null)
+            {
+                if (functionDescriptor.FilterAttributes != null)
+                {
+                    filters.AddRange(functionDescriptor.FilterAttributes);
+                }
+            }
+
+            // Check for class filters
+            if (functionDescriptor != null)
+            {
+                if (functionDescriptor.DeclaringFilterAttributes != null)
+                {
+                    filters.AddRange(functionDescriptor.DeclaringFilterAttributes);
+                }
+            }
+
+            return filters;
+        }
+
+        private static async Task InvokeExecutingFilters(List<IFunctionInvocationFilter> filters, ParameterHelper parameterHelper,
+            Dictionary<string, object> properties, CancellationTokenSource functionCancellationTokenSource, 
+            IFunctionInstance instance, ILogger logger, JobHost jobHost, JobHostConfiguration config)
+        {
+            if (filters.Any())
+            {
+                // Create the context
+                FunctionExecutingContext context = new FunctionExecutingContext(instance.Id, instance.FunctionDescriptor.FullName, parameterHelper.GetParametersAsDictionary(), properties, logger);
+                context.JobHost = jobHost;
+                context.Config = config;
+
+                foreach (var filter in filters)
+                {
+                    await filter.OnExecutingAsync(context, functionCancellationTokenSource.Token);
+                }
+            }
+        }
+
+        private static async Task InvokeExecutedFilters(List<IFunctionInvocationFilter> filters, ParameterHelper parameterHelper,
+            Dictionary<string, object> properties, CancellationTokenSource functionCancellationTokenSource,
+            IFunctionInstance instance, ILogger logger, JobHost jobHost, JobHostConfiguration config, FunctionResult functionResult)
+        {
+            if (filters.Any())
+            {
+                // Create the context
+                FunctionExecutedContext context = new FunctionExecutedContext(instance.Id, instance.FunctionDescriptor.FullName, parameterHelper.GetParametersAsDictionary(), properties, logger, functionResult);
```

Comment:
yes, this should be ShortName. 

---
In reply to: [128640456](https://github.com/Azure/azure-webjobs-sdk/pull/1244#discussion_r128640456) [](ancestors = 128640456)